src/move/emojicoin_dot_fun/sources/emojicoin_dot_fun.move

```diff
@@ -932,6 +1009,30 @@ module emojicoin_dot_fun::emojicoin_dot_fun {
         (result as u64)
     }
 
+    inline fun tvl_clamm(
+        virtual_reserves: Reserves,
+    ): u128 {
+        let quote_virtual = virtual_reserves.quote;
+        let quote_real = quote_virtual - QUOTE_VIRTUAL_FLOOR;
+        if (quote_real == 0) { // Report no TVL if no one has bought into bonding curve.
+            0
+        } else {
+            let base_virtual = virtual_reserves.base;
+            // Determine total amount of all base still locked in the market.
+            let base_real = base_virtual - BASE_VIRTUAL_FLOOR + EMOJICOIN_REMAINDER;
```

Comment:
You add the `EMOJICOIN_REMAINDER` here because it's technically `base` that's locked, right? Feel free to resolve this yourself if my understanding is correct

Reply:
Correct, during the bonding curve state, there is `base_virtual - BASE_VIRTUAL_FLOOR` locked in the bonding curve CLAMM, and `EMOJICOIN_REMAINDER` set aside to for the eventual CPAMM state